tests.py

```diff
@@ -458,6 +458,30 @@ def test_multi_ifo_fails(self):
                 **self.signal_parameters,
             )
 
+    def test_makefakedata_usecached(self):
+
+        # Make everything from scratch
+        writer = self.writer_class_to_test(
+            outdir=self.outdir,
+            **default_Writer_params,
+            **default_signal_params,
+            **default_transient_params,
+        )
+        writer.make_data(verbose=True)
+        first_time = os.path.getmtime(writer.sftfilepath)
+
+        # Re-run, and should be unchanged
+        writer.make_data(verbose=True)
+        second_time = os.path.getmtime(writer.sftfilepath)
+        self.assertTrue(first_time == second_time)
+
+        # third run: touch the .cff to force regeneration
+        time.sleep(1)  # make sure timestamp is actually different!
+        os.system("touch {}".format(writer.config_file_name))
+        writer.run_makefakedata()
+        third_time = os.path.getmtime(writer.sftfilepath)
+        self.assertFalse(first_time == third_time)
+
```

Comment:
This is the problematic test, where executables are different for some reason.

Reply:
Actual command line calls being compared. Seems like the order of the arguments is different...?

```
15:04 INFO    : Here is the CL from the old one:
lalapps_Makefakedata_v4 --outSingleSFT=TRUE --outSFTbname="TestData/H-4_H1_1800SFT_test-700000000-7200.sft" --IFO="H1" --ephemEarth="/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz" --ephemSun="/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz" --startTime=700000000 --duration=7200 --fmin=29.96404307059555 --Band=0.07191313888519033 --Tsft=1800 --window="tukey" --tukeyBeta=0.001 --h0=5 --cosi=0 --phi0=0 --Freq=30 --noiseSqrtSh=1 --lineFeature=TRUE --randSeed=42
15:04 INFO    : Here is the CL from the one to be created:
lalapps_Makefakedata_v4 --lineFeature=TRUE --outSingleSFT=TRUE --outSFTbname="TestData/H-4_H1_1800SFT_test-700000000-7200.sft" --IFO=H1 --noiseSqrtSh="1" --window="tukey" --tukeyBeta=0.001 --startTime=700000000 --duration=7200 --fmin=29.96404307059555 --Band=0.07191313888519033 --Tsft=1800 --h0 5.0 --Freq 30.0 --phi0 0 --cosi=0 --ephemEarth="/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz" --ephemSun="/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz" --randSeed=42
```

Reply:
Apparently, [XLALUserVarGetLog](https://lscsoft.docs.ligo.org/lalsuite/lal/_user_input_8c_source.html#l01217) is used to get such a log before putting it into the SFT, and the order seems to be consistent with that specified in [MFDv4's XLALInitUserVars](https://lscsoft.docs.ligo.org/lalsuite/lalapps/makefakedata__v4_8c.html#a66b8a22a0d05442022448b06c66a5512).

So I think I'll simply re-sort the way in which arguments are concatenated.
Also, it looks like that was the exact approach taken in other MFDv4-based classes, such as `FrequencyModulatedArtifactWriter`.

Reply:
@dbkeitel I'll need a yes/no from you on this last point.

Reply:
I have no clear memory of that and would need to dig into the code. But I'm pretty sure that somewhere we already have a set-based, i.e. order-independent, solution to it...

Reply:
`helper_functions.match_commandlines()`, is that not used here?

Reply:
You are right, `helper_functions.match_commandlines`, thanks.

Reply:
Paying actual attention:
Old cl: `--h0=5  --Freq=30 --phi0=0 --noiseSqrtSh=1`
New cl: `--h0 5.0 --Freq 30.0 --phi0 0 --noiseSqrtSh="1"`